### PR TITLE
Learn Page - Initial work hooking up API to client

### PR DESCRIPTION
**Pull request by indirectlylit**
* added new state and pages for learn mode - content and root
* added some filters to the api so we only have media content, not topics

### CODE REVIEW

File: kolibri/content/api.py [hunk 1]
```diff
@@ -33,18 +33,18 @@ def title_description_filter(self, queryset, value):
...
+        descendants = recc_node.get_descendants(include_self=False).exclude(kind__in=['topic', ''])
+        siblings = recc_node.get_siblings(include_self=False).exclude(kind__in=['topic', ''])
         data = descendants | siblings  # concatenates different querysets
         return data
 
     def filter_recommendations(self, queryset, value):
         # if ContentInteractionLog.objects.count() == 0:
-        content_ids = queryset.values_list('content_id', flat=True)
+        pks = queryset.values_list('pk', flat=True).exclude(kind__in=['topic', ''])
         count = queryset.count()
```
Comment (ralphiee22) [inline kolibri/content/api.py]: change this line to
`count = pks.count()`,
and the tests should pass.

File: kolibri/content/api.py [hunk 2]
```diff
@@ -33,18 +33,18 @@ def title_description_filter(self, queryset, value):
...
         data = descendants | siblings  # concatenates different querysets
         return data
 
     def filter_recommendations(self, queryset, value):
         # if ContentInteractionLog.objects.count() == 0:
-        content_ids = queryset.values_list('content_id', flat=True)
-        count = queryset.count()
+        pks = queryset.values_list('pk', flat=True).exclude(kind__in=['topic', ''])
+        count = pks.count()
         if count > 100:
```
Comment (66eli77) [inline kolibri/content/api.py]: do you want to change it to 25 also?

File: kolibri/plugins/learn/assets/src/actions.js [hunk 3]
```diff
@@ -67,7 +67,8 @@ function showExploreContent(store, id) {
 
   Resources.getModel(id).fetch()
     .then((attributes) => {
-      const pageState = _contentState(attributes);
+      const pageState = {};
```
Comment (66eli77) [inline kolibri/plugins/learn/assets/src/actions.js]: wouldn't _contentState( ) guarantee to return an object? Is there other purpose to declare the object this way?

File: kolibri/content/api.py [hunk 4]
```diff
@@ -33,18 +33,18 @@ def title_description_filter(self, queryset, value):
...
-        descendants = recc_node.get_descendants(include_self=False)
-        siblings = recc_node.get_siblings(include_self=False)
+        descendants = recc_node.get_descendants(include_self=False).exclude(kind__in=['topic', ''])
+        siblings = recc_node.get_siblings(include_self=False).exclude(kind__in=['topic', ''])
         data = descendants | siblings  # concatenates different querysets
         return data
 
     def filter_recommendations(self, queryset, value):
         # if ContentInteractionLog.objects.count() == 0:
-        content_ids = queryset.values_list('content_id', flat=True)
```
Comment (66eli77) [inline kolibri/content/api.py:43]: isn't the queryset here include every content node in the database? if so, that's gonna be an unnecessary huge list

File: kolibri/plugins/learn/assets/src/actions.js [hunk 5]
```diff
@@ -77,8 +78,40 @@ function showExploreContent(store, id) {
 }
 
 function showLearnRoot(store) {
+  store.dispatch('SET_LOADING');
   store.dispatch('SET_PAGE_NAME', PageNames.LEARN_ROOT);
-  store.dispatch('SET_PAGE_STATE', {}); // TODO
+
+  Resources.getCollection({ recommendations: '' }).fetch()
+    .then((children) => {
```
Comment (66eli77) [inline kolibri/plugins/learn/assets/src/actions.js]: I think currently recommendations will return random nodes, not necessary the children, the naming might mislead people

Comment (66eli77) [inline kolibri/plugins/learn/assets/src/actions.js]: oh, relative to root, any random node is consider root's child.

File: kolibri/content/api.py (hunk 4 above)
Comment (indirectlylit) [inline kolibri/content/api.py:43]: Yes - any thoughts on a more efficient way of randomly selecting 25 content nodes?

**Comment by indirectlylit** [comment]
thanks @66eli77 - great feedback. I made updates based on your comments. If you have any thoughts on a more efficient way of selecting random nodes, let @ralphiee22 know :)

**Pull request merged**